hil-test/tests/uart.rs

```diff
@@ -81,47 +86,23 @@ mod tests {
         // send/receive with some other common baud rates to ensure this is
         // working as expected. We will also using different clock sources
         // while we're at it.
-
-        #[cfg(not(feature = "esp32s2"))]
-        {
-            #[cfg(not(any(feature = "esp32", feature = "esp32c3", feature = "esp32c2")))]
-            {
-                // 9600 baud, RC FAST clock source:
-                ctx.uart.change_baud(9600, ClockSource::RcFast);
-                ctx.uart.write(7).ok();
-                let read = block!(ctx.uart.read());
-                assert_eq!(read, Ok(7));
-            }
-
-            // 19,200 baud, XTAL clock source:
-            #[cfg(not(feature = "esp32"))]
-            {
-                ctx.uart.change_baud(19_200, ClockSource::Xtal);
-                ctx.uart.write(55).ok();
-                let read = block!(ctx.uart.read());
-                assert_eq!(read, Ok(55));
-            }
-
-            // 921,600 baud, APB clock source:
-            ctx.uart.change_baud(921_600, ClockSource::Apb);
-            ctx.uart.write(253).ok();
-            let read = block!(ctx.uart.read());
-            assert_eq!(read, Ok(253));
-        }
-
-        #[cfg(feature = "esp32s2")]
-        {
-            // 9600 baud, REF TICK clock source:
-            ctx.uart.change_baud(9600, ClockSource::RefTick);
-            ctx.uart.write(7).ok();
-            let read = block!(ctx.uart.read());
-            assert_eq!(read, Ok(7));
-
-            // 921,600 baud, APB clock source:
-            ctx.uart.change_baud(921_600, ClockSource::Apb);
-            ctx.uart.write(253).ok();
+        let configs = [
```

Comment:
Nice.

Reply:
I got a bit fed up with nested `cfg(not)` :D